src/sessions/database_session.ts

```diff
@@ -0,0 +1,214 @@
+// deno-lint-ignore-file no-explicit-any
+import { Api } from "../tl/api.js";
+import { MemorySession } from "./memory_session.ts";
+import { AuthKey } from "../crypto/authkey.ts";
+import { Buffer, denodb } from "../../deps.ts";
+import { getPeerId } from "../utils.ts";
+import { returnBigInt } from "../helpers.ts";
+import { bigInt } from "../../deps.ts";
+
+// Avoiding this export in deps.ts
+import type { ConnectorOptions } from 'https://deno.land/x/denodb@v1.0.39/lib/connectors/connector.ts'
+
+const DataTypes = denodb.DataTypes;
+
+class Session extends denodb.Model {
+  static timestamps = true;
+
+  static fields = {
+    name: { type: DataTypes.STRING, primaryKey: true },
+    dcId: DataTypes.STRING,
+    port: DataTypes.INTEGER,
+    serverAddress: DataTypes.STRING,
+    authKey: DataTypes.BINARY,
+  }
+}
+
+class Entity extends denodb.Model {
+  static timestamps = true;
+
+  static fields = {
+    id: { type: DataTypes.STRING, primaryKey: true },
+    hash: { type: DataTypes.INTEGER, allowNull: false },
+    name: DataTypes.STRING,
+    username: DataTypes.STRING,
+    phone: DataTypes.STRING,
+  }
+}
+
+export enum DatabaseType {
+  Postgres,
+  MySQL,
+  MariaDB,
+  SQLite,
+  MongoDB
+}
+
+export interface ConnectionOptions<T extends ConnectorOptions> {
```

Comment:
I don't see where the `T` is used.

Reply:
It was being used because I was trying to make the whole thing completely type safe. I would like to have it so that if you pass in the Postgres DatabaseType as the adapter, the adapter options would know that they should be using the PostgresConnector options.

But I'm not there yet in my TypeScript knowledge

Reply:
I would like to have type safety as much as possible. So, I could give it a try.